### PR TITLE
Git storage

### LINKED ISSUE #38: Cookbook - write a tutorial for using git as a data source
We should have a 1-page tutorial showing how someone can set up a local git repo as their data source. The code samples should be copy-pasteable.

Prerequisites:
This tutorial should assume that someone has done the quickstart and the Card Guides, and should not re-explain the content from those areas.

To-do:
- [x] write an outline (bullet points) with the steps someone has to go through to create the files
- [x] add code samples to the outline that are copy-pasteable
- [x] create a branch on the template repo that has the completed demo
- [x] Get a reviewer of the code samples/steps to see if any can be simplified
- [x] Write the prose. A cookbook article should be fairly short. Here's an example of a  good cookbook article https://vue-vixens-workshop.netlify.com/workshop/nanos/nano2.html#get-started
- [x] submit a PR as soon as most of the tutorial is sketched out. It doesn't have to be perfect/complete
- [x] get a reviewer and iterate
- [ ] double check that code samples work, and add a link to the finished code
- [ ] ship it!

### PR DESCRIPTION
**Pull request by jenweber**
Closes #38

### CODE REVIEW

File: guides/data/git.md [hunk 1]
```diff
@@ -1,8 +1,134 @@
...
+
+Now that the repository is set up, we need to configure the plugin to
+use it. You will need to know the absolute path to your data repository.
+
+```bash
+pwd
+# prints /the/path/to/your/project-data/
+```
+
+In `cardhost/data-sources/default.js` for your project, set your default data source to git:
```
Comment (bagby) [inline guides/data/git.md]: should be `cardhost/cardstack/data-sources/default.js`

File: guides/data/git.md [hunk 2]
```diff
@@ -1,8 +1,134 @@
...
+
+```javascript
+{
```
Comment (bagby) [inline guides/data/git.md]: we might want to mention that there is an existing [default data source](https://github.com/cardstack/project-template/blob/master/cardhost/cardstack/data-sources/default.js#L72) with type `@cardstack/ephemeral`, and that this should *replace* that one.

Comment (bagby) [reply guides/data/git.md]: ...or maybe it's assumed that they already know that...?

File: guides/data/git.md [hunk 3]
```diff
@@ -1,8 +1,134 @@
...
-git has many advantages, including versioning, the ability to compare current and past states, and powerful rollback capabilities.
+In this guide, we will show how to save Card data to git by default.
+
+## Installing and configuring
+
+Install [`@cardstack/git`](https://github.com/cardstack/cardstack/tree/master/packages/git) in the `cardhost` of your project:
+
+```sh
+cd cardhost
+yarn add --dev @cardstack/git
```
Comment (bagby) [inline guides/data/git.md]: shouldn't this be `yarn install --dev @cardstack/git`? When I did `yarn add` it complained about `@cardstack/git` not being installed when I was trying to run `ember generate card foo` to generate a card.

File: guides/data/git.md [hunk 4]
```diff
@@ -1,8 +1,134 @@
...
+    }
+  }
+```
+
+*Never ever commit the file containing your private key.* If someone has your private key, they could delete anything on your GitHub, deploy malicious code, and other bad things.
+
+Always use an environmental variable to provide your key. You can do this from the command line:
+
+```bash
+GIT_PRIVATE_KEY="/path/to/your/private/key/id_rsa"
```
Comment (bagby) [inline guides/data/git.md]: is this supposed to be the path to the private key, or the private key itself? (I think it's the latter)

File: guides/data/git.md [hunk 5]
```diff
@@ -1,8 +1,134 @@
...
+
+First, create a repository on GitHub and initialize it with a `README.md`:
+
+![screenshot showing the README option checked](/images/github-create-repo.png)
+
+Make sure to choose "private" if you don't want to expose your data with the world. Choose "public" if you want to share your work with the open source community.
+
+After you have created the repository, copy the `ssh` URL, such as `git@github.com:my-username/project-data.git`.
+
+Now in `cardhost/data-sources/default.js`, configure your project to use the remote repository. Make sure to replace `your-username` with your GitHub username or organization name:
```
Comment (bagby) [inline guides/data/git.md]: should be `cardhost/cardstack/data-sources/default.js`

File: guides/data/git.md (hunk 2 above)
Comment (jenweber) [reply guides/data/git.md]: I like to spell out the assumptions anyway. Added!

**Comment by jenweber** [comment]
I removed the tutorial update from this PR and put it into another PR so we can merge it in after the initial movielist publishing.

**Pull request merged**